articles/server-platforms/aspnet-core/04-storing-tokens.md

```diff
@@ -0,0 +1,90 @@
+
+The names of the tokens will be saved as a semicolon-separated list inside the ".TokenNames" property. You will need to extract those names, and then for each of them extract the value of the token from the Properties.
+
+The value will be stored as a property with the name ".Token." suffixed with the name of the token. So, for example the "access_token" will be stored in a property with the name ".Token.access_token". 
+
+Once you have retrieved the value of the token, you can then simply store it as a claim for the `ClaimsIdentity`:
+
+```csharp
+app.UseOpenIdConnectAuthentication(new OpenIdConnectOptions("Auth0")
+{
+    // Set the authority to your Auth0 domain
+    Authority = $"https://{auth0Settings.Value.Domain}",
+
+    // Configure the Auth0 Client ID and Client Secret
+    ClientId = auth0Settings.Value.ClientId,
+    ClientSecret = auth0Settings.Value.ClientSecret,
+
+    // Do not automatically authenticate and challenge
+    AutomaticAuthenticate = false,
+    AutomaticChallenge = false,
+
+    // Set response type to code
+    ResponseType = "code",
+
+    // Set the callback path, so Auth0 will call back to http://localhost:60856/signin-auth0 
+    // Also ensure that you have added the URL as an Allowed Callback URL in your Auth0 dashboard 
+    CallbackPath = new PathString("/signin-auth0"),
+
+    // Configure the Claims Issuer to be Auth0
+    ClaimsIssuer = "Auth0",
+
+    // Saves tokens to the AuthenticationProperties
+    SaveTokens = true,
+
+    Events = new OpenIdConnectEvents()
+    {
+        OnTicketReceived = context =>
+        {
+            // Get the ClaimsIdentity
+            var identity = context.Principal.Identity as ClaimsIdentity;
+            if (identity != null)
+            {
+                // Check if token names are stored in Properties
+                if (context.Properties.Items.ContainsKey(".TokenNames"))
+                {
+                    // Token names a semicolon separated
+                    string[] tokenNames = context.Properties.Items[".TokenNames"].Split(';');
+
+                    // Add each token value as Claim
+                    foreach (var tokenName in tokenNames)
+                    {
+                        // Tokens are stored in a Dictionary with the Key ".Token.<token name>"
+                        string tokenValue = context.Properties.Items[$".Token.{tokenName}"];
+
+                        identity.AddClaim(new Claim(tokenName, tokenValue));
+                    }
+                }
+            }
+
+            return Task.FromResult(0);
+        }
+    }
+});
+```
+
+The `access_token` will now be stored as a claim called "access_token", so to retrieve it inside a controller you can simply use `User.Claims.FirstOrDefault("access_token").Value`
+
+::: warning-banner Beware of your cookie size
```

Comment:
This looks like that:

![image](https://cloud.githubusercontent.com/assets/11715799/16909026/65133cc0-4cd8-11e6-9337-353cd8739ead.png)

It looks like a bit too much, and the `ClaimsIdentity` is barely visible.